### PR TITLE
audit check_shim_references: Mirror keg_contain?

### PR DESCRIPTION
**Pull request by sjackman**
Do not report matches in binary files to files that do not exist.

- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/HEAD/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/HEAD/Library/Homebrew/test/PATH_spec.rb).
- [x] Have you successfully run `brew style` with your changes locally?
- [x] Have you successfully run `brew tests` with your changes locally?

-----

This PR mirrors the logic of `brew bottle`.
https://github.com/Homebrew/brew/blob/7db771be3072660d00b5bee96d39e65884e3a1a0/Library/Homebrew/dev-cmd/bottle.rb#L150-L151

It prevents false positives of matches to informational lines such as 
```
C++ Compiler: /usr/local/Homebrew/Library/Homebrew/shims/mac/super/clang++ 10.0.1.10010046
```

### CODE REVIEW

**Comment by Bo98** [comment]
While I do think that how it is in `bottle.rb` is problematic in some cases (it fails to detect `pkg-config` as non-relocatable), I agree that the behaviour should at least be consistent.

I wonder if the exact line matching logic should only apply to binary files however. We do want to catch cases where shell scripts have stuff like:

`/usr/local/Homebrew/Library/Homebrew/shims/mac/super/pkg-config --libs whatever`

I think ImageMagick is an example of this.

**Comment by sjackman** [comment]
> I wonder if the exact line matching logic should only apply to binary files however.

That makes sense to me. I've implemented that logic.

File: Library/Homebrew/formula_cellar_checks.rb [hunk 1]
```diff
@@ -207,12 +207,28 @@ def check_shim_references(prefix)
...
-      match = f.relative_path_from(keg.to_path)
+    keg.each_unique_file_matching(HOMEBREW_SHIMS_PATH) do |file|
+      relative_file = file.relative_path_from(keg.to_path)
 
-      next if match.to_s.match? %r{^share/doc/.+?/INFO_BIN$}
+      next if relative_file.to_s.match? %r{^share/doc/.+?/INFO_BIN$}
 
-      matches << match
+      result = if file.binary_executable? || file.dylib?
+        Utils.popen_read("strings", "-", file.to_s) do |io|
```
Comment (MikeMcQuaid) [inline Library/Homebrew/formula_cellar_checks.rb:216]: If this logic is essentially shared with `keg_contain?` can they both call a shared function?

**Comment by MikeMcQuaid** [comment]
> While I do think that how it is in `bottle.rb` is problematic in some cases (it fails to detect `pkg-config` as non-relocatable), I agree that the behaviour should at least be consistent.

I think if we've got a known problem in `bottle.rb` we should fix it there rather than making this consistent(ly incorrect).

**Pull request closed without merging**